### DISCUSSION TITLE
Sort topics by number of upvotes

### DISCUSSION THREAD

**Discussion by Zeugma440** [community/community#3328] [Discussions] [answered]
I'm considering using Discussions to maintain a public list of ideas that would receive my community's votes.

Every release, I'd catch the top 10 upvoted ideas and consider putting them on the next release.

But right now, to get that top 10, I have to browse all discussion pages. A "sort by most upvoted" would be warmly welcome !

**Comment by joel-ekstep** [comment]
Would the tab on top of the board 'Top <duration filter' help with this? I'm assuming that's the intent for that option.

**Reply by Zeugma440** [reply]
To be honest, I don't understand how that one works. Choosing "Past week" and "Past month" seems indeed to sort by upvotes, but choosing "All" or "Past year" gives a seemingly unordered result.

Is that a bug, or is it sorting by something else ?

**Reply by markseuffert** [reply]
I would be interested too. [For example here are unanswered discussions](https://github.com/datenstrom/yellow/discussions?discussions_q=sort%3Atop+is%3Aunanswered), where sorting by upvotes would be helpful. At the moment we don't have many unanswered discussions, so I can look trough them manually. It would be lovely to see what our community has upvoted most and then use this information to prioritise what need to be done.  

![Screen Shot](https://user-images.githubusercontent.com/2429165/118236152-05c7b400-b496-11eb-8436-26a4a63d5cc3.png)

As far as I can see, `sort:top is:unanswered` doesn't sort by community upvotes.

**Comment by PranamBhat** [comment]
Hey! 

Choosing "Past week", "Past month", "Past year" and "All" will gives you a list of discussions based on the "**Vote Counts**".

Here is the snapshot for your reference:


![GitHub Discussions Past week](https://user-images.githubusercontent.com/32120033/108022415-85d0b000-7046-11eb-95c8-05b8e637ad23.JPG)
![GitHub Discussions Past year](https://user-images.githubusercontent.com/32120033/108022423-8701dd00-7046-11eb-9a4b-649d0cc998bb.JPG)

**Reply by Zeugma440** [reply]
Not sure we understand each other here. My issue is a basic one : as an end user, I can't sort the topics by vote counts. Showing me HTML code talks to my inner developer but still doesn't solve the initial problem.

Here's a more visual way of showing what I mean

![image](https://user-images.githubusercontent.com/29414549/108034731-3bdfce00-7036-11eb-8a47-4b56c975230e.png)

You can clearly see vote counts being all mangled up on the first two pictures (all / past year), and the sorting order being correct on the rightmost picture (minus some of the displayed 4's being actually 5's when you look at the `upvoted-vote-count` property, but you get the general idea...)

**Reply by rijkvanzanten** [reply]
> Choosing "Past week", "Past month", "Past year" and "All" will gives you a list of discussions based on the "Vote Counts".

That doesn't sound right to me. Unless it's a bug in GH, sorting by `sort:top` definitely isn't based on the upvotes on our end:

<img width="1305" alt="CleanShot 2021-02-17 at 13 20 30@2x" src="https://user-images.githubusercontent.com/9141017/108249385-02d15780-7123-11eb-8075-29ae251f76e5.png">

(https://github.com/directus/directus/discussions?discussions_q=sort%3Atop)

---

Update 2/23/2021:

It seems to only work on "Past Month" and newer, but not on "All" or "Past Year"

**Reply by rijkvanzanten** [reply]
@becca What is `sort:top` supposed to sort on? It seems to sort by upvotes when choosing "Today" or "Past Week", but it seems to sort arbitrarily for any date ranges bigger than that 🤔 Is there a `sort:upvotes` equivalent in the works?

**Reply by jasonvarga** [reply]
> What is sort:top supposed to sort on?

https://github.com/github/feedback/discussions/3706#discussioncomment-610657

**Comment by michael-hawker** [comment]
Surprised this wasn't here too, was just trying to find the most voted things for the discussions beta feedback itself...

**Comment by HDVinnie** [comment]
Sort by upvotes is a must to help see what end-users are most interested in topic wise.

**Comment by TheMakerTales** [comment]
Is there any further development going into this? I'm a little surprised that this was not a quick fix?
This little feature would make the whole discussion section so much more valuable for open source communities. I was really expecting the functionality of [right click select blender community](https://blender.community/c/rightclickselect/?sorting=top&page=1&developmentStatus=open&category=modeling)

**Comment by mjonas87** [comment]
Completely agree...it seems like it _should_ be a simple feature. We just started using this to track requests for things to be documented. Had no idea I couldn't _sort_ by upvotes.

**Comment by ghost** [comment]
@Zeugma440 Hi! I had an idea similar to the proposal here! My idea would be to have an order the topics by the most or least commented: https://github.com/github/feedback/discussions/16617 - What do you think of this idea? is it a good idea or not?

**Comment by rowlandm** [comment]
![Screen Shot 2022-06-26 at 3 19 57 pm](https://user-images.githubusercontent.com/420050/175800560-736a7401-ea6e-4495-b29a-cf5d2dd83e23.png)

We are having the same issue, using top doesn't sort by upvotes.

**Comment by jayceeday** [comment, marked as answer]
Hello everyone and thanks for the feedback!

This has now been updated so you can go try it out.

You can click Top: (timeframe) and it will order the discussions by number of upvotes!

<img width="1492" alt="Screenshot 2022-08-31 at 16 19 28" src="https://user-images.githubusercontent.com/86061938/187701530-50dbd6fa-36c2-4760-aa98-d3f0271ee581.png">

**Reply by Zeugma440** [reply]
Looks super fine to me now. Thanks for considering my request !

**Reply by markseuffert** [reply]
I get some unexpected results, for example [here are some higher voted entries towards the end](https://github.com/datenstrom/yellow/discussions?discussions_q=sort%3Atop&page=20). I would have expect 1 or 0 votes at the end of the list. You can also reproduce something similar in this repository, for example go to `Discussions` start page, click `Top: All`, scroll down to the pagination, go to the last page.

**Reply by mjonas87** [reply]
Amaaaaaazing!!!! ❤️ 

We've been tracking a number of things in Discussions and using upvotes is going to allow us to focus on what matters most to our Engineers. Thank you so much! 🙌

**Reply by usulpro** [reply]
Unfortunately, this still doesn't work as expected :(
I've tried to select a sorting from the list or write sorting manually - same result
![image](https://user-images.githubusercontent.com/14885189/188059676-db6ecb01-ca64-40e8-8c3a-9368e7c28209.png)

**Reply by Zeugma440** [reply]
Looks like there's still some work to do. @markseuffert's example is a good one to focus on.

**Reply by jayceeday** [reply]
Thank you again for all your feedback and patience!

My apologies - there were still some bugs before when I said we have fixed this. We have now fixed those and it should now sort by most upvotes correctly. 

Please try it out and let me know if you have any more feedback or problems!

Thanks again 💗

**Reply by markseuffert** [reply]
Thank you for the update. I can confirm it works as expected. 👍

**Comment by davidfstr** [comment]
Although this feature has technically been implemented, I would say it is difficult to discover, because the dropdown that implements the sorting is misleadingly labeled as if it's already performing the sorting when it's not. See screenshot below for a better explanation:
![Screenshot 2024-08-22 at 12 58 00 PM](https://github.com/user-attachments/assets/57ea3ea7-c5d6-4664-96f6-6430edf4e136)

**Comment by zachgoll** [comment]
Is there any way to set an "organization default" so that Github users who click on our Discussion categories can see a view that we've pre-configured for them?

For example, we'd like to create a "Top Feature Requests" list that users can see without having to create the sort themselves:

![CleanShot 2024-10-23 at 18 09 14](https://github.com/user-attachments/assets/bad32bab-825f-4a1d-b784-3732b8e4afc2)
